sympy/core/tests/test_function.py

```diff
@@ -777,6 +777,14 @@ def test_diff_wrt_not_allowed():
     assert diff(exp(x*y), x*y, 0) == exp(x*y)
 
 
+def test_diff_wrt_intlike():
+    class Two:
+        def __int__(self):
```

Comment:
Do we want to test for a class with `_sympy_` too? How about a class added to `converter` ?

Reply:
Yes, maybe. At least this particular test would have picked up on this particular case. We also have the SAGE integration tests and that's the only known application for this but I thought it would be good to add something in the main test suite.

Reply:
Fair enough, I guess we can't put in tests for every possible situation.